route/build/src/tracer.f90

```diff
@@ -0,0 +1,364 @@
+           zc        => rch_param%SIDE_SLOPE, & ! channel side slope
+           zf        => rch_param%FLDP_SLOPE, & ! floodplain slope
+           bankVol   => rch_param%R_STORAGE,  & ! bankful volume
+           L         => rch_param%RLENGTH)      ! channel length
+
+ if (.not. isHW .or. hw_drain_point==top_reach) then
+
+   allocate(Cprev(nMole), stat=ierr, errmsg=cmessage)
+   if(ierr/=0)then; message=trim(message)//trim(cmessage); return; endif
+
+   ! initialize previous time step flow
+   Cprev(1:nMole) = subrch_state%CC     ! flow state at previous time step
+
+   if (verbose) then
+     write(iulog,'(A,1X,G12.5)') ' length [m] =',L
+   end if
+
+   ! time-step adjustment so Courant number is less than 1
+   dTsub = dt/ntSub
+
+   if (verbose) then
+     write(iulog,'(A,1X,I3,A,1X,G12.5)') ' No. sub timestep=',nTsub,' sub time-step [sec]=',dTsub
+   end if
+
+   allocate(Clocal(1:nMole, 0:1), stat=ierr, errmsg=cmessage)
+   if(ierr/=0)then; message=trim(message)//trim(cmessage); return; endif
+
+   do it = 1, nTsub
+     ! use constant flow velocity
+     Qbar = sum(subrch_state%Q)/nMole
+     if (abs(Qbar)>1.e-50_dp) then
+       Ybar = flow_depth(abs(Qbar), bt, zc, S, n, zf=zf, bankDepth=bankDepth)
+       Abar = flow_area(abs(Ybar), bt, zc, zf=zf, bankDepth=bankDepth)
+       Vbar =  Qbar/Abar
+     else
+       Vbar=0._dp
+     end if
+     dk = 0._dp   ! dk=0 -> no diffusion, only advection
+     call solve_ade(L,                  & ! input: river parameter data structure
+                    nMole,              & ! input: number of sub-segments
+                    dTsub,              & ! input: time_step [sec]
+                    Cupstream,          & ! input: quantity from upstream [unit of quantity]
+                    Vbar,               & ! input: reach mean flow velocity [m/s]
+                    dk,                 & ! input: diffusivity [m2/s]
+                    Clat,               & ! input: lateral quantity into chaneel [unit of quantity]
+                    Cprev,              & ! input: quantity at previous time step [unit of quantity]
+                    Clocal,             & ! inout: quantity soloved at current time step [unit of quantity]
+                    verbose,            & ! input: reach index to be examined
+                    ierr,message)
+     if(ierr/=0)then; message=trim(message)//trim(cmessage); return; endif
+   end do
+
+   ! For very low flow condition, outflow - inflow may exceed current storage, so limit outflow and adjust flow profile
+   max_outCC=rflux%ROUTE(idxRoute)%reach_solute_flux/dt + Clocal(1,1)
+   if (Clocal(nMole-1,1)>max_outCC) then
+     Clocal(2:nMole,1) = max_outCC*0.999
+   end if
+   rflux%ROUTE(idxRoute)%reach_solute_flux = rflux%ROUTE(idxRoute)%reach_solute_flux + (Cupstream - Clocal(nMole-1,1))*dt
+
+   rflux%ROUTE(idxRoute)%reach_solute_flux = Clocal(nMole-1,1) + Clat
+   ! store final outflow in data structure
```

Comment:
I suggest removing commented out code.

Reply:
removed these as some cleanup of this subroutine